app/models/katello/errata_status.rb

```diff
@@ -9,17 +10,30 @@ def self.status_name
       N_("Errata")
     end
 
+    def errata_status_installable
+      Setting[:errata_status_installable]
+    end
+
+    def profiles_reporter_package_name
+      if host.content_facet.host_tools_installed?
+        Katello::Host::ContentFacet::HOST_TOOLS_PACKAGE_NAME
+      else
+        Katello::Host::ContentFacet::SUBSCRIPTION_MANAGER_PACKAGE_NAME
+      end
+    end
+
     def to_label(_options = {})
-      installable = Setting[:errata_status_installable]
       case status
       when NEEDED_SECURITY_ERRATA
-        installable ? N_("Security errata installable") : N_("Security errata applicable")
+        errata_status_installable ? N_("Security errata installable") : N_("Security errata applicable")
       when NEEDED_ERRATA
-        installable ? N_("Non-security errata installable") : N_("Non-security errata applicable")
+        errata_status_installable ? N_("Non-security errata installable") : N_("Non-security errata applicable")
       when UP_TO_DATE
         N_("All errata applied")
+      when UNREGISTERED
+        N_("Could not calculate errata status because the host is not registered.")
       when UNKNOWN
-        N_("Could not calculate errata status, ensure host is registered and the katello-host-tools package is installed")
+        N_("Could not calculate errata status because %s has not reported any package or repository information." % profiles_reporter_package_name)
```

Comment:
I think this is very valuable and i've wanted to add something to help 'debug' applicability issues, this is a good first step.  

One issue though is that this saying that it hasn't reported package or repo information, but thats not actually what we're checking.  I think it would be more valuable to report specifically that 'no package' or 'no repo' information has been reported based off the actual contents of those tables.

Reply:
@jlsherrill Updated. I think I captured what you meant although the wording is a little nuanced. Let me know if it's what you were thinking / clear enough.